docs/advanced-features/open-telemetry.md

```diff
@@ -0,0 +1,107 @@
+---
+description: Learn how instrument your Next.js app with OpenTelemetry.
+---
+
+> **Note**: This feature is experimental, you need to explicitly opt-in by providing `experimental.instrumentationHook = true;` in your `next.config.js`.
+
+# OpenTelemetry in Next.js
+
+Observability is crucial for understanding and optimizing the behavior and performance of your Next.js app.
+
+As applications become more complex, it becomes increasingly difficult to identify and diagnose issues that may arise. By leveraging observability tools, such as logging and metrics, developers can gain insights into their application's behavior and identify areas for optimization. With observability, developers can proactively address issues before they become major problems and provide a better user experience. Therefore, it is highly recommended to use observability in your Next.js applications to improve performance, optimize resources, and enhance user experience.
+
+We will use terms like _Span_, _Trace_ or _Exporter_ throughout this doc, all of which can be found in [the OpenTelemetry Observability Primer](https://opentelemetry.io/docs/concepts/observability-primer/).
+
+> **Note:** We currently support observability bindings only in serverless functions.
+> We don't provide any for `edge` or client side code.
+
+## Getting Started
+
+Next.js supports OpenTelemetry, which is platform agnostic. You can change your observability provider without changing your code.
+Read [Official OpenTelemetry docs](https://opentelemetry.io/docs/) for more information about OpenTelemetry and how it works.
+
+To get started, you must install the required packages:
+
+```bash
+npm install @opentelemetry/api @opentelemetry/exporter-trace-otlp-http @opentelemetry/resources @opentelemetry/sdk-trace-base @opentelemetry/semantic-conventions
+```
+
+Next, create a custom [`instrumentation.ts`](./instrumentation.md) file in the root of the project.
+Since you can't import OpenTelemetry on `edge` we recommend creating second file, that get's only imported in correct environment.
```

Comment:
```suggestion
Since you can't import OpenTelemetry in `register` functions executed on the Edge Runtime, we recommend creating second file that will only be imported if the runtime is `nodejs`.

The following example demonstrates an `instrumenation.ts` file that checks the runtime, then imports the code it needs to run:
```

Reply:
@jankaifer What do you think of this change?